packages/kit/src/exports/index.js

```diff
@@ -1,12 +1,13 @@
 import { HttpError, Redirect, ActionFailure } from '../runtime/control.js';
 import { BROWSER, DEV } from 'esm-env';
 
-// For some reason we need to type the params as well here,
-// JSdoc doesn't seem to like @type with function overloads
 /**
- * @type {import('@sveltejs/kit').error}
- * @param {number} status
- * @param {any} message
+ * Creates an `HttpError` object with an HTTP status code and an optional message.
+ * This object, if thrown during request handling, will cause SvelteKit to
+ * return an error response without invoking `handleError`.
+ * Make sure you're not catching the thrown error, which would prevent SvelteKit from handling it.
+ * @param {number} status The [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses). Must be in the range 400-599.
+ * @param {{ message: string } extends App.Error ? App.Error | string | undefined : never} message An object that conforms to the App.Error type. If a string is passed, it will be used as the message property.
```

Comment:
added `@overload`

Reply:
ah nvm figured it out, it was unrelated — i had removed `.ts` files from `tsconfig.build.json`

Reply:
You need the overload because the type definition is:
- if you did not modify `App.Error`, in other words only `{ message: string }` is required, then the second parameter is optional (because it is falling back to `Error: ${status}` in that case)
- if you did modify `App.Error`, in other words more than just `{ message: string }` is requreid, then you can't omit the second parameter and is required

You can only define this with overloads

Reply:
gotcha, i missed the fact that one was optional and one wasn't. updated